src/main/java/org/opentosca/csarrepo/servlet/AbstractServlet.java

```diff
@@ -62,37 +64,59 @@ protected abstract void doGet(HttpServletRequest request, HttpServletResponse re
 		return root;
 	}
 
+	/**
+	 * @return the base path of the repository <code>/csarrepo</code>
+	 */
 	public String getBasePath() {
 		return this.getServletContext().getContextPath();
 	}
 
-	public User checkAuthentication(HttpServletRequest request, HttpServletResponse response) throws IOException {
-		HttpSession session = request.getSession(false);
-
-		// Session management for user interface
-		if (null == session && null != request.getAttribute("user") && !(request.getAttribute("user") instanceof User)) {
-			LOGGER.info("Check of authentication failed: Session is null!");
-			response.sendRedirect(LoginServlet.PATH);
-		} else {
-			return (User) request.getAttribute("user");
+	/**
+	 * Triggers authentication over HTTP
+	 * 
+	 * @param request
+	 *            The incoming request for the servlet
+	 * @param response
+	 *            The outgoing response of the servlet
+	 * @return The user
+	 * @throws IOException
+	 */
+	public User checkBasicAuthentication(HttpServletRequest request, HttpServletResponse response) throws IOException {
+		// Basic authentication for REST API
+		if (null != request.getHeader("Authorization")) {
+			String header = request.getHeader("Authorization");
+			if (header.substring(0, 6).equals("Basic ")) {
+				String[] credentials = Base64.decodeAsString(header.substring(6).getBytes()).split(":");
+				if (credentials.length > 1) {
+					String username = credentials[0];
+					String password = credentials[1];
+					String hashedPassword = Hash.sha256(password);
+					LoadCheckedUserService loadCheckedUserService = new LoadCheckedUserService(username, hashedPassword);
+					if (!loadCheckedUserService.hasErrors()) {
+						return loadCheckedUserService.getResult();
+					}
```

Comment:
@przytads @TKosch don't we want to generate log entries for failed invocations of loadCheckedUserService?